project-files/Stylesheets/JS/main.js

```diff
@@ -1,24 +1,76 @@
-    document.getElementById("canonicalization-method").innerHTML = xmlDoc.getElementsByTagName("ds:CanonicalizationMethod")[0].getAttribute("Algorithm")
-    document.getElementById("signature-timestamp").innerHTML = xmlDoc.getElementsByTagName("xades:SigningTime")[0].childNodes[0].nodeValue
-    document.getElementById("signature-timestamp-utc").innerHTML = xmlDoc.getElementsByTagName("xades:SigningTime")[0].childNodes[0].nodeValue
-    document.getElementById("signer-certificate").innerHTML = x509['organizationName-X520'] + " " + x509['organizationalUnitName-X520']
-    document.getElementById("signer-name").innerHTML = x509['commonName-X520']
-    document.getElementById("signer-name2").innerHTML = x509['commonName-X520']
-    document.getElementById("signer-time").innerHTML = x509['UTCTime']
-    document.getElementById("signer-time2").innerHTML = x509['UTCTime']
-    document.getElementById("signer-computer-time-utc").innerHTML = xmlDoc.getElementsByTagName("xades:SigningTime")[0].childNodes[0].nodeValue
+  var path = window.location.pathname;
+  if (!path.endsWith(".xml")) {
+    return
+  }
+  var parser = new DOMParser();
+  const page = await (await fetch(path)).text();
+  var xmlDoc = parser.parseFromString(page, "text/xml");
+  var x509Cert = xmlDoc.getElementsByTagName("ds:X509Certificate")[0].childNodes[0].nodeValue
+  x509 = decodeCert(x509Cert)
+  console.log(x509)
+  document.getElementById("signature-method").innerHTML = xmlDoc.getElementsByTagName("ds:SignatureMethod")[0].getAttribute("Algorithm")
+  document.getElementById("canonicalization-method").innerHTML = xmlDoc.getElementsByTagName("ds:CanonicalizationMethod")[0].getAttribute("Algorithm")
+  document.getElementById("signature-timestamp").innerHTML = xmlDoc.getElementsByTagName("xades:SigningTime")[0].childNodes[0].nodeValue
+  document.getElementById("signature-timestamp-utc").innerHTML = xmlDoc.getElementsByTagName("xades:SigningTime")[0].childNodes[0].nodeValue
+  document.getElementById("signer-certificate").innerHTML = x509['organizationName-X520'] + " " + x509['organizationalUnitName-X520']
+  document.getElementById("signer-name").innerHTML = x509['commonName-X520']
+  document.getElementById("signer-name2").innerHTML = x509['commonName-X520']
+  document.getElementById("signer-time").innerHTML = x509['UTCTime']
+  document.getElementById("signer-time2").innerHTML = x509['UTCTime']
+  document.getElementById("signer-computer-time-utc").innerHTML = xmlDoc.getElementsByTagName("xades:SigningTime")[0].childNodes[0].nodeValue
 }
 
-getXMLCertificateInfo()
\ No newline at end of file
+async function updateOnlineStatus() {
+  let status = document.getElementById("connectionStatus");
+  let validateButton = document.getElementById("validateButton");
+  let connectionText = document.getElementById("connectionText");
+
+  if (navigator.onLine) {
```

Comment:
> For some reason, I don't seem to get this working in Mac OS X / Safari.  If I stop the eIDAS server docker image, the loaded certificate still states that the connection is online, but the validate button does not work, of course.
> 
> Maybe we need to take a Zoom on this?

Yes, let's take a Zoom on this. I already know what is causing problems with other browsers than Chrome. Let's discuss if fixing the problem is possible in the current implementation.

Reply:
> Furthermore, if the web page is loaded before the eIDAS server is up, I think should should show initially OFFLINE, when the eIDAS server is still starting, then dynamically change to online.
> 
> I presume it would be best to open this as a separate story / bug.

I agree. Currently the offline status is fetched from the browsers .isonline() function. Let's create another story so I can implement small ping to eIDAS to determine the offline functionality.